proposed/template-renderer.md

```diff
@@ -0,0 +1,207 @@
+using the Interface to give or configure the `template` used by the template renderer. This can be achieved with any
+type of injection or configuration.
+
+Basic implementor implementation:
+
+```php
+<?php
+
+use Psr\TemplateRenderer\TemplateRendererInterface;
+
+class TemplateRenderer implements TemplateRendererInterface
+{
+    public function render(string $template, array $context = []): string
+    {
+        switch ($template) {
+            case 'home.tpl';
+
+                return $this->renderTemplate('<p>{{name}}</p>', $context);
+            default: 
+                throw new TemplateNotFoundException($template);
+        }
+    }
+    
+    /**
+     * @param array<string, mixed> $context
+     * @return string
+     */
+    private function renderTemplate(string $content, array $context): string
+    {
+        foreach ($context as $key => $value) {
+            $content = str_replace('{{' . $key . '}}', $context[$key], $content);
+        }
+        
+        return $content;
+    }
+}
+
+use Psr\TemplateRenderer\TemplateNotFoundExceptionInterface;
+
+class TemplateNotFoundException extends \RuntimeException implements TemplateNotFoundExceptionInterface
+{
+    public function __construct(private string $template, ?\Throwable $previous = null)
+    {
+        parent::__construct('Template not found: "' . $template . '"', 0 , $previous);
+    }
+    
+    public function getTemplate(): string
+    {
+        return $this->template;
+    }
+}
+```
+
+Basic user implementation:
+
+```php
+<?php
+
+use Psr\TemplateRenderer\TemplateRendererInterface;
+
+class SomeController
+{
+    public function __construct(
+        private TemplateRendererInterface $templateRenderer,
+        private string $template,
+    )
+
+    public function someAction(): string
+    {
+        $context = [/* load something */]; 
+    
+        return $this->templateRenderer->render($this->template, $context);
+    }
+}
+```
+
+Example enduser implementation A:
+
+```php
+<?php
+
+$controller = new Controller($twig, '@Context/pages/mail.html.twig');
+$controller->someAction();
+```
+
+Example enduser implementation B:
+
+```php
+<?php
+
+$controller = new Controller($blade, 'mail');
+$controller->someAction();
+```
+
+Example enduser implementation C:
+
+```php
+<?php
+
+$controller = new Controller($latte, 'mail.latte');
+$controller->someAction();
+```
+
+## Interfaces
+
+### TemplateRendererInterface
+
+```php
+<?php
+
+namespace Psr\TemplateRenderer;
+
+interface TemplateRendererInterface {
+    /**
+     * Render the template with the given context data.
+     *
+     * @param string $template
+     * @param array<string, mixed> $context
+     *
+     * @return string
+     */
+    public function render(string $template, array $context = []): string;
+}
+```
+
+### TemplateNotFoundExceptionInterface
+
+```php
+<?php
+
+namespace Psr\TemplateRenderer;
+
+interface TemplateNotFoundExceptionInterface
```

Comment:
```suggestion
interface TemplateNotFoundExceptionInterface extends InvalidArgumentException
```

Reply:
Interface cannot be extended from a class

Reply:
> Interface cannot be extended from a class

Oh, sorry. Fix to `Throwable`.

Reply:
Make sense thx for the suggestion.